Update config spec to include new fields (#4477)
pachyderm/pachyderm: doc/docs/master/reference/config_spec.md

kubernetes contexts (and in fact reference the kubernetes context that they're
associated with.)

#### Source

> An int specifying where the config came from. This is for internal use, and

Reviewer: ```suggestion
An integer that specifies where the config came from. This parameter is for internal use only and
```

> shouldn't be modified.

Reviewer: ```suggestion
should not be modified.
```

The currently active transaction for batching together pachctl commands. This
can be set or cleared via many of the `pachctl * transaction` commands.

#### Cluster name

> The name of the underlying kubernetes cluster, extracted from the kubernetes

Reviewer: ```suggestion
The name of the underlying Kubernetes cluster, extracted from the Kubernetes
```

context.

#### Auth info

> The name of the underlying kubernetes cluster's auth credentials, extracted

Reviewer: ```suggestion
The name of the underlying Kubernetes cluster's auth credentials, extracted
```

> from the kubernetes context.

Reviewer: ```suggestion
from the Kubernetes context.
```

#### Namespace

> The underlying kubernetes cluster's namespace, extracted from the kubernetes

Reviewer: ```suggestion
The underlying Kubernetes cluster's namespace, extracted from the Kubernetes
```

context.

#### Cluster ID

> The pachyderm cluster ID - used to ensure we're running operations on the

Reviewer: ```suggestion
The pachyderm cluster ID that is used to ensure the operations run on the
```

expected cluster.

#### Port forwarders

> A mapping of service name -> local port. This field is populated when running

Reviewer: ```suggestion
A mapping of `service name -> local port`. This field is populated when you run
```

explicit port forwarding (`pachctl port-forward`), so that subsequent
`pachctl` operations know to use the explicit port forwarder.

> This field will be removed when the `pachctl port-forward` operation

Reviewer: ```suggestion
This field is removed when the `pachctl port-forward` operation
```

> completes, but note that you may need to manually delete the field from your

Reviewer: ```suggestion
completes. You might need to manually delete the field from your
```

> config if the process was killed being being able to do so.

Reviewer: ```suggestion
config if the process failed to remove the field automatically.
```
